natss/pkg/dispatcher/dispatcher.go

```diff
@@ -267,7 +253,7 @@ func (s *SubscriptionsSupervisor) UpdateSubscriptions(channel *messagingv1alpha1
 			continue
 		}
 		// subscribe and update failedSubscription if subscribe fails
-		natssSub, err := s.subscribe(cRef, subRef)
+		natssSub, err := s.subscribe(context.TODO(), cRef, subRef)
```

Comment:
and then this gets a real context. 

At the very least, you should use `context.Background()` because that gets the sig-terms